core/org.wso2.carbon.ui/src/main/java/org/wso2/carbon/ui/CarbonUILoginUtil.java

```diff
@@ -487,7 +488,12 @@ protected static boolean handleLogin(CarbonUIAuthenticator authenticator,
             log.debug("Authentication failure ...", e);
             try {
                 request.getSession().invalidate();
-                getAuthenticator(request).unauthenticate(request);
+                CarbonUIAuthenticator carbonUIAuthenticator = getAuthenticator(request);
+                if (carbonUIAuthenticator == null && request.getParameterMap().containsKey(SAML_RESPONSE_KEY)) {
+                    response.sendRedirect(contextPath + "/carbon/sso-acs/authFailure.jsp");
```

Comment:
let's use a const to the error page context

Reply:
Fixed with [348e086](https://github.com/wso2/carbon-kernel/pull/3428/commits/348e086391f061c63572a329066ba74bed1e4670)